Library/Homebrew/tap.rb

```diff
@@ -141,11 +168,21 @@ def uninstall
     unpin if pinned?
     formula_count = formula_files.size
     Descriptions.uncache_formulae(formula_names)
+    unlink_manpages
     @path.rmtree
     @path.dirname.rmdir_if_possible
     puts "Untapped #{formula_count} formula#{plural(formula_count, "e")}"
   end
 
+  def unlink_manpages
+    return unless (path/"man").exist?
+    (path/"man").find do |src|
+      next if src.directory?
+      dst = HOMEBREW_PREFIX/src.relative_path_from(path)
+      dst.delete if dst.exist?
```

Comment:
We need to check the dst is a symlink pointed to this tap. Also we need to remove dir if possible.